### PR TITLE
refactor(spanview): Extract RelativeBar component from TraceSpanView

### PR DESCRIPTION
**Pull request by yurishkuro**
## Which problem is this PR solving?

This is a focused extraction from #3348 (by @chinmay3012), isolating the reusable `RelativeBar` component so it can be reviewed and merged independently.

## Description of the changes

- Add `RelativeBar` component (`components/common/RelativeBar.tsx` + `.css`) — a simple proportional bar that renders a fill relative to a max value, using CSS design tokens for theming (light/dark mode compatible). Original implementation by @chinmay3012.
- Refactor `TraceSpanView` duration column to use `<RelativeBar>` instead of the equivalent inline div structure.

The remaining changes from #3348 (RelativeBar integration into TraceStatistics, dark mode fixes, performance improvements) are left for the original PR.

## How was this change tested?

- All existing tests pass (`npm test`)

## Checklist

- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have run lint and test steps successfully

### CODE REVIEW

File: packages/jaeger-ui/src/components/common/RelativeBar.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  value: number;
+  maxValue: number;
+};
+
+const RelativeBar: React.FC<Props> = ({ value, maxValue }) => {
+  const percentage = (value / (maxValue || 1)) * 100;
+
+  return (
+    <div className="RelativeBar">
+      <div className="RelativeBar--fill" style={{ width: `${Math.max(percentage, 2)}%` }} />
```
[the remark this answers is not in the document]
Comment (yurishkuro) [reply packages/jaeger-ui/src/components/common/RelativeBar.tsx:19]: Fixed in ba4ae014 — guarded against non-positive/non-finite maxValue and clamped percentage to [0, 100] before the minimum fill logic.

File: packages/jaeger-ui/src/components/common/RelativeBar.css [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  border-radius: 5px;
+  overflow: hidden;
+  border: 1px solid var(--border-strong, rgba(255, 255, 255, 0.1));
+}
+
+[data-theme='dark'] .RelativeBar {
+  background: rgba(255, 255, 255, 0.1);
+  border-color: rgba(255, 255, 255, 0.2);
+}
+
```
[the remark this answers is not in the document]
Comment (yurishkuro) [reply packages/jaeger-ui/src/components/common/RelativeBar.css]: Fixed in 3313a02e — removed the [data-theme='dark'] block; the CSS variables are already theme-aware.

File: packages/jaeger-ui/src/components/TracePage/TraceSpanView/index.tsx [hunk 3]
```diff
@@ -158,26 +158,7 @@ export default function TraceSpanView(props: Props) {
...
-                <div
-                  style={{
-                    width: `${Math.max(percentage, 2)}%`,
-                    height: '100%',
-                    background: 'var(--interactive-primary)',
-                    borderRadius: '2px',
-                  }}
-                />
-              </div>
+              <RelativeBar value={span.duration} maxValue={maxDuration} />
```
[the remark this answers is not in the document]
Comment (yurishkuro) [reply packages/jaeger-ui/src/components/TracePage/TraceSpanView/index.tsx:161]: Fixed in 54199789 — removed the dead .duration-bar-background dark-mode rule from TraceSpanView/index.css.

File: packages/jaeger-ui/src/components/common/RelativeBar.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (yurishkuro) [reply packages/jaeger-ui/src/components/common/RelativeBar.tsx]: Added in e9e9a571 — 7 unit tests covering proportional fill, 100% cap, 2% minimum, zero/negative/non-finite maxValue edge cases.

File: packages/jaeger-ui/src/components/common/RelativeBar.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  maxValue: number;
+};
+
+const RelativeBar: React.FC<Props> = ({ value, maxValue }) => {
+  const safeMax = Number.isFinite(maxValue) && maxValue > 0 ? maxValue : 1;
+  const percentage = Math.min(100, Math.max(0, (value / safeMax) * 100));
+
+  return (
+    <div className="RelativeBar">
+      <div className="RelativeBar--fill" style={{ width: `${Math.max(percentage, 2)}%` }} />
```
[the remark this answers is not in the document]
Comment (yurishkuro) [reply packages/jaeger-ui/src/components/common/RelativeBar.tsx:19]: Fixed in 82561f47 — normalize non-finite `value` to 0 before computing percentage, same pattern as `maxValue`. Added a test for NaN.

**Comment by chinmay3012** [comment]
Thank you for the update and for the mention! I'm glad to see the RelativeBar component being utilized across other parts of the UI.

I have just pushed the remaining updates to my PR (#3348) which includes the full integration into Trace Statistics, the performance optimizations, and the dark mode styling fixes. I've also ensured all unit tests are passing with the newly extracted component. Ready for your final review whenever you have a moment!"
